src/transformers/modeling_utils.py

```diff
@@ -2955,6 +2956,35 @@ def register_for_auto_class(cls, auto_class="AutoModel"):
 
         cls._auto_class = auto_class
 
+    def to_bettertransformer(self) -> "PreTrainedModel":
+        """
+        Converts the model to use PyTorch's BetterTransformer through the [Optimum
+        integration](https://huggingface.co/docs/optimum/bettertransformer/overview).
+
+        Returns:
+            `PreTrainedModel`: The model converted to BetterTransformer.
+        """
+        if not is_optimum_available():
+            raise ImportError("The package `optimum` is required to use BetterTransformer.")
+
+        from optimum.bettertransformer import BetterTransformer
```

Comment:
I think a version check on optimum with a clear error message would be good? Also can the transform be applied twice? If not there should be a check and a clear error message as well.

Reply:
> Also can the transform be applied twice? If not there should be a check and a clear error message as well.

I think there's currently no check for this @younesbelkada . I will add it on the Optimum side. So as to keep the transformers side as lightweight as possible.